auction-app-frontend/src/components/ProfileTab/ProfileTab.jsx

```diff
@@ -1,36 +1,197 @@
+
+            let month = '';
+            let day = '';
+            let year = '';
+
+            if (response.dob) {
+                const dob = new Date(response.dob);
+                month = dob?.getMonth() + 1;
+                day = dob?.getDate();
+                year = dob?.getFullYear();
+            }
+
+            reset({
+                firstName: response.firstName,
+                lastName: response.lastName,
+                email: response.email,
+                street: response.paymentInfoEntity?.address,
+                city: response.paymentInfoEntity?.city,
+                country: response.paymentInfoEntity?.country,
+                zipCode: response.paymentInfoEntity?.zipCode,
+                nameOnCard: response.paymentInfoEntity?.creditCardEntity?.nameOnCard,
+                cardNumber: response.paymentInfoEntity?.creditCardEntity?.cardNumber,
+                expirationMonth: expMonth,
+                expirationYear: expYear,
+                month: month,
+                day: day,
+                year: year
+            });
+          })
+          .catch((error) => {
+            setError(error);
+            setLoading(false);
+          })
+          .finally(() => {
+            setLoading(false);
+          });
+    };
+
+    const updateUserData = (data) => {
+        setUpdateLoading(true);
+
+        // null fields because some fields are optional
+        const userData = {
+            firstName: data.firstName,
+            lastName: data.lastName,
+            email: data.email,
+            dob: data.year && data.month && data.day ? new Date(data.year, data.month - 1, data.day) : null,
+            address: data.street || null,
+            city: data.city || null,
+            country: data.country || null,
+            zipCode: data.zipCode || null,
+            nameOnCard: data.nameOnCard || null,
+            cardNumber: data.cardNumber || null,
+            expirationDate: data.expirationYear && data.expirationMonth ? new Date(data.expirationYear, data.expirationMonth - 1, 1) : null
+        };    
+
+        setUpdateLoading(true);
+
+        updateUser(userId, userData)
+            .then((response) => {
+                setUpdateLoading(false);
+            }).catch((error) => {
+                setUpdateError(error.message);
+            });
+    };
+
+    const updateUserProfileImage = () => {
+        const formData = new FormData();
+
+        formData.append("image", file);
+
+        setUpdateLoading(true);
+
+        updateProfileImage(userId, formData)
+            .then((response) => {
+                setUpdateLoading(false);
+            }).catch((error) => {
+                setUpdateError(error.message);
+            });
+    };
+    
+    useEffect(() => {
+        if (userId) fetchUserInformation();
+    }, [userId, reset]);
+
     const onSubmit = (data) => {
-        console.log(data);
+        updateUserData(data);
     }
 
+    const changeProfilePicture = () => {
```

Comment:
As I see here, this method only opens modal, so change the name of the method to match its purpose